tests/src/ut_test_control_plane.c

```diff
@@ -0,0 +1,169 @@
+/*
+ * If not stated otherwise in this file or this component's LICENSE file the
+ * following copyright and licenses apply:
+ *
+ * Copyright 2023 RDK Management
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* Standard Libraries */
+#include <stdlib.h>
+#include <string.h>
+#include <stdbool.h>
+#include <assert.h>
+#include <signal.h>
+#include <unistd.h>
+
+/* Module Includes */
+#include <ut.h>
+#include <ut_log.h>
+
+#include <ut_control_plane.h>
+
+static UT_test_suite_t *gpAssertSuite1 = NULL;
+static UT_test_suite_t *gpAssertSuite2 = NULL;
+
+static ut_controlPlane_instance_t *gInstance = NULL;
+
+void wait_for_1_second()
+{
+    sleep(1);
+}
+
+void test_ut_control_testInitExit()
+{
+    ut_controlPlane_instance_t *pInstance = NULL;
+    ut_controlPlane_instance_t *pInstance1 = NULL;
+
+    UT_ControlPlane_Exit(NULL);
+    UT_ASSERT( pInstance == NULL );
+
+    pInstance = UT_ControlPlane_Init(8080);
+    UT_ControlPlane_Start(pInstance);
```

Comment:
your testing `test_ut_control_testInitExit()` - you should not be testing Start() that' a seperate test. Where you no longer need to test Init Exit() you can be rest assured that it works by testing it first.